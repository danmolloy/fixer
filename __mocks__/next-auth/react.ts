import { mockSession } from '../session';

module.exports = {
  __esModule: true,
  ...jest.requireActual('next-auth/react'),
  auth: jest.fn(() => {
    return mockSession;
  }),
  signIn: jest.fn(),
  signOut: jest.fn(),
  useSession: jest.fn(),
};
