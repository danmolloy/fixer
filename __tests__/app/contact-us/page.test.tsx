import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import ContactPage from '../../../app/contact-us/page';

describe('<ContactInfo />', () => {
  beforeEach(() => {
    render(<ContactPage />);
  });
  it('contact-page is in the document', () => {
    const contactPage = screen.getByTestId('contact-page');
    expect(contactPage).toBeInTheDocument();
  });
  it('contact-info is in the document', () => {
    const contactInfo = screen.getByTestId('contact-info');
    expect(contactInfo).toBeInTheDocument();
  });
  it('contact-form is in the document', () => {
    const contactForm = screen.getByTestId('contact-form');
    expect(contactForm).toBeInTheDocument();
  });
  //it("matches snapshot", () => {})
});
