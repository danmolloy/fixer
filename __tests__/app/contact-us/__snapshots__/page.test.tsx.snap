// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactInfo /> matches snapshot 1`] = `
<div
  data-testid="contact-page"
>
  <div
    class="px-4 py-6"
    data-testid="contact-info"
  >
    <h1
      class="font-bold"
    >
      Contact us
    </h1>
  </div>
  <div
    class="p-4"
    data-testid="contact-form"
  >
    <form
      action="#"
      class="flex flex-col"
    >
      <div
        class="flex w-full flex-col py-4"
        data-testid="name-input-div"
      >
        <label
          class=""
          for="name-input"
        >
          Name
        </label>
        <input
          class="my-1 h-8 max-w-[60vw] rounded border px-1 shadow-sm undefined"
          data-testid="name-input-input"
          id="name-input"
          label="Name"
          max="30"
          name="name"
          type="text"
          value=""
        />
      </div>
      <div
        class="flex w-full flex-col py-4"
        data-testid="email-input-div"
      >
        <label
          class=""
          for="email-input"
        >
          Email
        </label>
        <input
          class="my-1 h-8 max-w-[60vw] rounded border px-1 shadow-sm undefined"
          data-testid="email-input-input"
          id="email-input"
          label="Email"
          max="45"
          name="email"
          type="email"
          value=""
        />
      </div>
      <div
        class="flex flex-col"
      >
        <label
          class="form-label "
          for="msg-text"
        >
          Message
           
        </label>
        <textarea
          class="w-full rounded-md border p-1 lg:w-[60vw] text-black shadow-sm"
          id="msg-text"
          maxlength="500"
          multiline="6"
          name="message"
          rows="4"
          type="textarea"
        />
        <div
          class="h-6"
        />
      </div>
      <button
        class="w-24 self-end rounded bg-blue-600 px-2 py-1 text-white shadow hover:bg-blue-500 "
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
